app/src/components/App/App.js

```diff
@@ -1,11 +1,196 @@
 import "./App.css";
-import { Route, Routes } from "react-router-dom";
+import { Navigate, Route, Routes, useNavigate } from "react-router-dom";
 import Main from "../Main/Main";
 import SavedNews from "../SavedNews/SavedNews";
 import React from "react";
+import ProtectedRoute from "../ProtectedRoute/ProtectedRoute";
+import CurrentUserContext from "../../contexts/CurrentUserContext";
+import api from "../../utils/NewsApi";
+import mainApi from "../../utils/MainApi";
 
 function App() {
+  const navigate = useNavigate();
+  const [isSignInFormOpened, setIsSignInformOpened] = React.useState(false);
+  const [isSignUpFormOpened, setIsSignUpformOpened] = React.useState(false);
+  const [signInState, setSignInState] = React.useState(true);
+  const [signUpState, setSignUpState] = React.useState(true);
   const [isMobileNavbarOpen, setIsMobileNavbarOpen] = React.useState(false);
+  const [isSuccessFormOpened, setIsSuccessformOpened] = React.useState(false);
+  const [isRegistered, setIsRegistered] = React.useState(false);
+  const [isLoggedIn, setIsLoggedIn] = React.useState(false);
+  const [currentUser, setCurrentUser] = React.useState({});
+  const [articles, setArticles] = React.useState([]);
+  const [maxArticlesRows, setMaxArticlesRows] = React.useState(1);
+  const [searchWorld, setSearchWorld] = React.useState("");
+  const [preloaderOpened, setPreloaderOpened] = React.useState(false);
+  const [savedNews, setSavedNews] = React.useState([]);
+  const [signUpError, setSignUpError] = React.useState("");
+  const [token, setToken] = React.useState(localStorage.getItem("token"));
+  const [errors, setErrors] = React.useState({
+    email: "",
+    password: "",
+    username: "",
+  });
+  const [data, setData] = React.useState({
+    email: "",
+    password: "",
+    username: "",
+  });
+
+  React.useEffect(() => {
+    if (token) {
+      mainApi
+        .getSavedArticles(token)
+        .then((articles) => {
+          if (articles) {
+            setSavedNews(articles.data);
+          }
+        })
+        .catch((err) => console.log(err));
+    }
+  }, [token, savedNews]);
```

Comment:
`token, savedNews`


please, delete `savedNews` from here

* There is a constant server request. SEe here http://joxi.ru/Y2LaM90txooRPA . The app freezes because of that


You get `savedNews` inside the `effect` and it causes recalling it again and again


```js

  }, []);

```